bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationConfigurationHandler.java

```diff
@@ -0,0 +1,143 @@
+                    .getAsJsonObject();
+            salt = authenticateJsonObject.get("salt").getAsString();
+
+            String saltedPassword = new StringBuilder(password).append(salt).toString();
+            MessageDigest mDigest = MessageDigest.getInstance("SHA1");
+
+            byte[] mDigestedPassword = mDigest.digest(saltedPassword.getBytes());
+            StringBuilder loginPostStringBuilder = new StringBuilder();
+            for (int i = 0; i < mDigestedPassword.length; i++) {
+                loginPostStringBuilder.append(Integer.toString((mDigestedPassword[i] & 0xff) + 0x100, 16).substring(1));
+            }
+            String saltedmDigestedPwd = Base64.getEncoder().encodeToString(mDigest.digest(saltedPassword.getBytes()));
+
+            String loginPostJsonData = "{\"mode\":1,\"userId\":\"" + username + "\",\"pwh\":\"" + saltedmDigestedPwd
+                    + "\"}";
+
+            Request loginPostJsonResponse = httpClient.POST(urlLogin + "?sessionId=" + sessionId);
+            loginPostJsonResponse.header(HttpHeader.CONTENT_TYPE, "application/json");
+            loginPostJsonResponse.content(new StringContentProvider(loginPostJsonData));
+            ContentResponse loginPostJsonDataContentResponse = loginPostJsonResponse.send();
+
+            String loginPostResponse = new String(loginPostJsonDataContentResponse.getContent());
+
+            JsonObject loginPostJsonObject = (JsonObject) new JsonParser()
+                    .parse(transformJsonResponse(loginPostResponse));
+
+            sessionId = extractSessionId(loginPostJsonObject);
+
+            // Part to sending data to Inverter
+            String postJsonData = "";
+
+            if (dxsId.contentEquals("16777984")) {
+                // Works with inverterName, name will be changed, due to "" around value
+                postJsonData = "{\"dxsEntries\":[{\"dxsId\":" + dxsId + ",\"value\":\"" + value + "\"}]}";
+            } else {
+                // Works not with inverterName, name will not be changed, due to "" around value, but the other
+                // configuration options will be changed.
+                postJsonData = "{\"dxsEntries\":[{\"dxsId\":" + dxsId + ",\"value\":" + value + "}]}";
+            }
+
+            Request postJsonDataRequest = httpClient.POST(url + "/api/dxs.json?sessionId=" + sessionId);
+            postJsonDataRequest.header(HttpHeader.CONTENT_TYPE, "application/json");
+            postJsonDataRequest.content(new StringContentProvider(postJsonData));
+
+            ContentResponse postJsonDataContentResponse = postJsonDataRequest.send();
+            String postResponse = new String(postJsonDataContentResponse.getContent());
+
+            JsonObject postJsonObject = (JsonObject) new JsonParser().parse(transformJsonResponse(postResponse));
+            sessionId = extractSessionId(postJsonObject);
+        } catch (JsonIOException e) {
+            logger.debug("Could not read the response.");
```

Comment:
You might want to include the exception's message.
```suggestion
            logger.debug("Could not read the response: {}", e.getMessage());
```